GTBitsOfGood/georgiaCORE
# Madeline/add auth users

## PR Title/Tagline

Issue Number(s): #20  

What does this PR change and why? PR adds authenticated users page to allow admins to alter who can access the app. PR also adds page protection so that only authorized users can view / and /navigation-editor and /admin. Also only Administrator roles can view the /admin page since that page gives access to edit users.

### Checklist


### Critical Changes

-New authUser model and actions 
-New /admin page
-/ and /navigation-editor and /admin are protected pages 

### Related PRs

- 

### How to Test

-Check if being logged out allows access to /, /navigation-editor, /admin (it shouldn't)
-Check if not having Administrator role allows access to /admin (it shouldn't)
-Check if adding/editing/deleting users makes the actions displayed on the page or at least on refresh
-Check if sorting sorts by email or role as desired
-Check if modals appear correctly
-Check if page buttons change view of data to different pages
